cql3/restrictions/statement_restrictions.cc

```diff
@@ -2748,14 +2749,27 @@ void statement_restrictions::prepare_indexed_global(const schema& idx_tbl_schema
         (*_idx_tbl_ck_prefix)[pos] = replace_column_def(e, &idx_tbl_schema.clustering_column_at(pos));
     }
 
+    if (std::ranges::any_of(*_idx_tbl_ck_prefix | std::views::drop(1), is_empty_restriction)) {
+        // If the partition key is not fully restricted, the index clustering key is of no use.
+        (*_idx_tbl_ck_prefix) = std::vector<expr::expression>();
+        return;
+    }
+
     add_clustering_restrictions_to_idx_ck_prefix(idx_tbl_schema);
 
+    auto pk_expressions = (*_idx_tbl_ck_prefix)
+            | std::views::drop(1)   // skip the token restriction
+            | std::views::take(_schema->partition_key_size()) // take only the partition key restrictions
+            | std::views::transform(expr::as<expr::binary_operator>) // we know it's an EQ
+            | std::views::transform(std::mem_fn(&expr::binary_operator::rhs)) // "solve" for the column value
+            | std::ranges::to<std::vector>();
```

Comment:
Nitpick: Personally, it would have been easier for me to read this code if it was a for loop, and not something in modern Reverse Polish Notation ;-)